controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -264,6 +264,14 @@ private SegmentRange convert(final String scope,
                 .thenApplyAsync(x -> x.stream().anyMatch(z -> z.segmentId() == segmentId), executor);
     }
 
+    public CompletableFuture<Boolean> isStreamCutValid(final String scope,
+                                                     final String stream,
+                                                     final Map<Long, Long> streamCut) {
```

Comment:
Do we want to gate this on our current streamCut discussions? as it seems like this method will need to be refactored significantly.

Reply:
No, this is introducing the call and not refactoring the validation. @shiveshr has explicitly separated adding the call from refactoring the validation.